### PR TITLE
decoder: enable low latency mode for AVC decoder

### PR DESCRIPTION
**Pull request by YuJiankang**
Signed-off-by: jkyu <jiankang.yu@intel.com>

### CODE REVIEW

File: decoder/vaapidecoder_h264.cpp [hunk 1]
```diff
@@ -1102,6 +1105,20 @@ bool VaapiDecoderH264::DPB::add(const PicturePtr& picture)
         compPicture->m_picStructure = VAAPI_PICTURE_FRAME;
     }
 
+    if (m_isLowLatencymode) {
+        while(true) {
+            it = find_if(m_pictures.begin(), m_pictures.end(), isOutputNeeded);
```
Comment (xuguangxin) [inline decoder/vaapidecoder_h264.cpp]: Could we add isLowLatencyOuptutFrame and combine
isOutputNeeded , and isFrame(*it) && (!m_isOutputStarted || ((*it)->m_poc -m_lastOutputPoc == 1) to the function.

File: decoder/vaapidecoder_h264.cpp [hunk 2]
```diff
@@ -336,6 +336,8 @@ VaapiDecoderH264::DPB::DPB(OutputCallback output)
     , m_maxFrameNum(0)
     , m_maxNumRefFrames(0)
     , m_maxDecFrameBuffering(H264_MAX_REFRENCE_SURFACE_NUMBER)
+    , m_isOutputStarted(false)
```
Comment (xuguangxin) [inline decoder/vaapidecoder_h264.cpp:339]: do we need reset this in flush?

**Review by xuguangxin — CHANGES_REQUESTED** [review]

File: decoder/vaapidecoder_h264.cpp [hunk 3]
```diff
@@ -1102,6 +1105,20 @@ bool VaapiDecoderH264::DPB::add(const PicturePtr& picture)
...
     }
 
+    if (m_isLowLatencymode) {
+        while(true) {
+            it = find_if(m_pictures.begin(), m_pictures.end(), isOutputNeeded);
+            if (it == m_pictures.end())
+                break;
+            if (isFrame(*it) && (!m_isOutputStarted || ((*it)->m_poc -m_lastOutputPoc == 1))) {
+                output(*it);
+                if (!m_isOutputStarted)
```
Comment (xuguangxin) [inline decoder/vaapidecoder_h264.cpp]: do not need this if and it slow than "m_isOutputStarted = true;" only

**Comment by YuJiankang** [comment]
updated

File: decoder/vaapidecoder_h264.cpp (hunk 2 above)
Comment (xuguangxin) [reply decoder/vaapidecoder_h264.cpp:339]: @YuJiankang , any update on this.
could you work with dongping to add some test case for low latency  mode

**Comment by uartie** [comment]
please rebase

File: decoder/vaapidecoder_h264.cpp [hunk 4]
```diff
@@ -1102,15 +1105,35 @@ bool VaapiDecoderH264::DPB::add(const PicturePtr& picture)
...
+    }
+}
+
 void VaapiDecoderH264::DPB::flush()
 {
     bumpAll();
     clearRefSet();
     m_pictures.clear();
     m_prevPicture.reset();
+    m_isOutputStarted = false;
```
Comment (xuguangxin) [inline decoder/vaapidecoder_h264.cpp:1136]: do we need update m_lastOutputPoc here?

**Review by xuguangxin — CHANGES_REQUESTED** [review]

**Review by xuguangxin — APPROVED** [review]

**Pull request merged**